httomo/task_runner.py

```diff
@@ -243,6 +245,28 @@ def run_tasks(
         # TODO: After the blocks loop is complete the full data
         # is in "data_full_section" and the hdf5 file can be saved
         # or resliced if needed
+
+        # Calculate stats on result of the last method in the section
+        #
+        # NOTE: Currently assuming that the dataset to calculate stats from is
+        # the same input dataset from the beginning of the sections loop; should
+        # this instead be getting the output dataset from the last method that
+        # was in the section which has just completed?
+        section.output_stats = min_max_mean_std(data_full_section, comm)
+
+        # Assume that, after every section has been completed (except for the
+        # last section), a reslice should occur
+        if i < len(platform_sections) - 1:
```

Comment:
I guess we currently assume that the reslice happens always at the end of each section? In the pipeline bellow 2 reslices happened while only 1 should be. I guess `should_reslice` previously controlled it? What do we have now instead to check for this?

```
Section 0
httomolibgpu.misc.corr
httomolibgpu.misc.corr
httomolibgpu.misc.corr
httomolibgpu.prep.normalize
Section 1
httomolibgpu.recon.rotation
httomolibgpu.prep.stripe
httomolibgpu.recon.algorithm
WARNING: Reslicing is performed more than once in this pipeline, is there a need for this?
Section 2
httomolib.misc.images
```

Reply:
May be slightly re-think the platform section stuff: 

- I feel that we need to reslice ONLY when we have a change of the pattern and do not reslice when the architecture changes (and the pattern remains). Would the section need to change in this case? Probably, yes.
- However for the CPU section we don't need to iterate over blocks, and methods can be decoupled and the results saved. _Should every CPU method become a separate section then?_ I know it is a bit drastic, but nothing really holds them together, they are separate entities memory-wise apart from the notion of pattern. However if we keep them in one section then the saving of the data become a bit messy... 
- To conclude may be we need to apply sectioning logic to GPU device-to-device only?